groovy-language-server/src/main/java/com/palantir/groovylanguageserver/GroovyTextDocumentService.java

```diff
@@ -91,7 +92,9 @@ public GroovyTextDocumentService(CompilerWrapperProvider provider) {
 
     @Override
     public CompletableFuture<List<? extends Location>> references(ReferenceParams params) {
-        throw new UnsupportedOperationException();
+        return CompletableFuture
+                .completedFuture(provider.get().findReferences(params).stream().map(symbol -> symbol.getLocation())
+                        .filter(location -> Ranges.isValid(location.getRange())).collect(Collectors.toList()));
```

Comment:
I think that should be the intended behavior. If you want to find references of `friend` then it should find all occurrences of this instance of `friend` in the project. This is currently not implemented, because we don't check expressions. As in `Animal anotherAnimal = friend` we don't recognize variable references right now. Also, if you give the position of the type `Animal` then the behavior also makes sense, giving all type references of `Animal`.